nimble_build_system/cad/fasteners.py

```diff
@@ -123,6 +164,26 @@ def __init__(
 
         self._length = length
 
```

Comment:
Thanks for the refactor here. Looks like a fair bit of work, but I think makes a lot of sense.